sanic_openapi/doc.py

```diff
@@ -119,19 +121,19 @@ def serialize(self):
         return {"type": "string", "format": "uuid", **super().serialize()}
 
 
-definitions = {}
-
-
 class Object(Field):
-    def __init__(self, cls, *args, object_name=None, **kwargs):
+    def __init__(
+        self, cls, *args, definitions: Optional[Dict] = None, object_name=None, **kwargs
+    ):
         super().__init__(*args, **kwargs)
 
         self.cls = cls
         self.object_name = object_name or cls.__name__
+        self.definitions = definitions or {}
 
-        register_as = object_name or "{}.{}".format(cls.__module__, cls.__qualname__)
-        if register_as not in definitions:
-            definitions[register_as] = (self, self.definition)
+        register_as = object_name or "{}".format(cls.__qualname__)
+        if register_as not in self.definitions:
+            self.definitions[register_as] = self.definition
```

Comment:
I think the root cause of `cars` does not work properly is here. In the original version, we use a global variable to store the `definitions`. And now, you remove this global variable but you didn't pass `self.definitions` to other variables (ex: `Spec`, `RouteSpec`, `RouteSpecs`...). But I'm not sure how to pass `self.definitions`. 🤔  Let's take some time to figure it out.

Reply:
Yeah, I added self.definitions to the `Swagger` class and thought it would be passed around from there but maybe I missed something. Will take another look at it tomorrow.

Reply:
I'm also not sure how to deal with this the best. I'd like to write some tests to catch it first too though. Any ideas on the best way to do that?